triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Server.scala

```diff
@@ -98,11 +103,18 @@ class Server(
 
   private def startTrigger(
       credentials: UserCredentials,
-      triggerName: Identifier): Either[String, JsValue] = {
+      triggerName: Identifier,
+      existingInstance: Option[UUID] = None): Either[String, JsValue] = {
     for {
       trigger <- Trigger.fromIdentifier(compiledPackages, triggerName)
-      triggerInstance = UUID.randomUUID
-      _ <- triggerDao.addRunningTrigger(RunningTrigger(triggerInstance, triggerName, credentials))
+      triggerInstance <- existingInstance match {
```

Comment:
The logic here feels like it might be in the wrong place. Maybe we should move the code for adding it to the table out of `startTrigger` instead of passing an `Option` in here?

Reply:
Not sure - the steps before and after are required in both cases. Will consider changing later though.